File: Problem Set 1/Cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void) 
{
    float dollars;
    int coins = 0, quarter = 25, dime = 10, nickel = 5, penny = 1;

    // Prompt user for an amount of change
    do
    {
        dollars = get_float("Change: ");
    } 
    while (dollars < 0);

    // Convert amount: dollars to cents in an interger, and round amount
    int amount = round(dollars * 100);

    // Use the largest coin possible and keep in track of coins used
    // While quarters can be used
    while (amount >= quarter)
    {
        // Increase count
        coins += 1;
        // Decrease amount by a quarter
        amount -= quarter;
    }
    // While dimes can be used
    while (amount >= dime)
    {
        coins += 1;
        amount -= dime;    
    }
    // While nickels can be used
    while (amount >= nickel)
    {
        coins += 1;
        amount -= nickel;    
    }
    // While pennies can be used
    while (amount >= penny)
    {
        coins += 1;
        amount -= penny;    
    }

    // Print the final number of coins
    printf("I have %i coin(s).\n", coins);
}

File: Problem Set 6/Cash/cash.py
from cs50 import get_float, get_int

def main():
    # Define variables
    coins = 0
    quarter = 25
    dime = 10
    nickel = 5
    penny = 1
    
    # Prompt user for an amount of change
    while True:
        dollars = get_float("Change: ")
        if dollars > 0:
            break
    
    # Convert amount: dollars to cents in an interger, and round amount
    amount = round(dollars * 100)
    
    # Use the largest coin possible and keep in track of coins used
    while (amount >= quarter):
        coins += 1
        amount -= quarter
    while (amount >= dime):
        coins += 1
        amount -= dime
    while (amount >= nickel):
        coins += 1
        amount -= nickel
    while (amount >= penny):
        coins += 1
        amount -= penny
    
    # Print the final number of coins
    print(f"I have {coins} coin(s).")

if __name__ == "__main__":
    main()
